src/errors/mod.rs

```diff
@@ -0,0 +1,35 @@
+use failure::Fail;
+
+/// Error states that could be returned from tarpaulin
+#[derive(Fail, Debug)]
+pub enum RunError {
+    /// Error in cargo manifests
+    #[fail(display="Failed to parse Cargo.toml! Error: {}", _0)]
+    Manifest(String),
+    /// Cargo failed to run
+    #[fail(display="Cargo failed to run! Error: {}", _0)]
+    Cargo(String),
+    /// Error trying to resolve package configuration in manifest
+    #[fail(display="Failed to resolve package configuration in manifest! Error: {}", _0)]
+    Packages(String),
+    /// Tests failed to compile
+    #[fail(display="Failed to compile tests! Error: {}", _0)]
+    TestCompileFailed(String),
+    /// Test failed during run
+    #[fail(display="Failed to run tests! Error: {}", _0)]
+    TestRuntimeFail(String),
+    /// Failed to parse
+    #[fail(display="Error while parsing: {}", _0)]
+    Parse(std::io::Error),
+    /// Failed to get test coverage
+    #[fail(display="Failed to get test coverage! Error: {}", _0)]
+    TestCoverage(String),
+    #[fail(display="{}", _0)]
+    IO(std::io::Error),
+}
```

Comment:
Please do take a good look at the error messages here, I guess there might be room for improvements